LocalyticsPlugin/Assets/Localytics/Localytics.cs

```diff
@@ -8,1073 +8,2163 @@
+			#endif
+		}
+
+		public static void RegisterForLocationEvents ()
+		{
+			#if UNITY_ANDROID
+			if (_locationListener == null)
+			{
+				_locationListener = new LocationListener();
+			}
+
+			LocalyticsClass.CallStatic("setLocationListener", _locationListener);
+			#elif UNITY_IOS
+			_registerReceiveLocationCallback (ReceiveLocationMessage);
+			#else
+			throw new NotImplementedException ("Localytics Unity SDK only supports iOS or Android");
+			#endif
+		}
+
+		public static void UnregisterForLocationEvents ()
+		{
+			#if UNITY_ANDROID
+			if (_locationListener == null)
+			{
+				return;
+			}
+
+			LocalyticsClass.CallStatic("setLocationListener", null);
+			#elif UNITY_IOS
+			_removeLocationCallback ();
+			#else
+			throw new NotImplementedException ("Localytics Unity SDK only supports iOS or Android");
+			#endif
+		}
+
+		// #################################################
+		// User Information
+		// #################################################
+
+		public static string CustomerId
+		{
+			get
+			{
+				#if UNITY_ANDROID
+				return LocalyticsClass.CallStatic<string>("getCustomerId");
+				#elif UNITY_IOS
+				return _customerId();
+				#else
+				throw new NotImplementedException ("Localytics Unity SDK only supports iOS or Android");
+				#endif
+			}
+			set
+			{
+				#if UNITY_ANDROID
+				LocalyticsClass.CallStatic("setCustomerId", value);
+				#elif UNITY_IOS
+				_setCustomerId(value);
+				#else
+				throw new NotImplementedException ("Localytics Unity SDK only supports iOS or Android");
+				#endif
+			}
+		}
+
+		public static string GetIdentifier (string key)
+		{
+			#if UNITY_ANDROID
+			return LocalyticsClass.CallStatic<string>("getIdentifier", key);
+			#elif UNITY_IOS
+			return _getIdentifier (key);
+			#else
+			throw new NotImplementedException ("Localytics Unity SDK only supports iOS or Android");
+			#endif
+		}
+
+		public static void SetIdentifier (string key, string value)
+		{
+			#if UNITY_ANDROID
+			LocalyticsClass.CallStatic("setIdentifier", key, value);
+			#elif UNITY_IOS
+			_setIdentifier (key, value);
+			#else
+			throw new NotImplementedException ("Localytics Unity SDK only supports iOS or Android");
+			#endif
+		}
+
+		public static void SetLocation (LocationInfo location)
+		{
+			#if UNITY_ANDROID
+			LocalyticsClass.CallStatic("setLocation", LocationInfoToLocation(location));
+			#elif UNITY_IOS
+			_setLocation(location.latitude, location.longitude);
+			#else
+			throw new NotImplementedException ("Localytics Unity SDK only supports iOS or Android");
+			#endif
+		}
+
+		// #################################################
+		// Developer Options
+		// #################################################
+
+		public static void SetOption (string key, string stringValue)
+		{
+			#if UNITY_ANDROID
+			LocalyticsClass.CallStatic("setOption", key, new AndroidJavaObject("java/lang/String", stringValue));
```

Comment:
Looks like we maybe have to do the same `AndroidJavaObject` stuff for strings too?  I'm seeing them used with and without the `AndroidJavaObject` all over the place in this file